content/blog/2019-05-24-sprint-report/contents.lr

```diff
@@ -0,0 +1,88 @@
+
+The new module supports migrating records but also any other index you may have in Elasticsearch such as event logging from usage statistics. The new module works by defining **migration recipes** and managing them via a CLI
+
+#### Demo
+
+First, you initialize a migration recipe (here a recipe named ``records``) which will create new indexes to hold the new data:
+```
+$ invenio index migration init "records"
+```
+
+Next, you kick-off the index migration:
+
+```
+$ invenio index migration run "records"
+```
+
+Index migration happens in two stages:
+- **Bulk migration:** First a snapshot of the current index is reindexing into the new index. Invenio-Index-Migrator is flexible an supports several methods. By default, we use the [Elasticsearch Reindex API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html) which in most cases are orders of magnitudes faster than any other method. Other methods allow you to use Invenio's default record indexing or simply implement your own.
+- **Synchronization:** Once the first snapshot has been fully indexed, a second job is started to keep the two indexes in sync. Essentially this works by indexing any modifications to the old index in the new one.
+
+At anypoint during the migration, you can check the status:
+
+```
+$ invenio index migration status "records"
+```
+
+Once you're ready, you can roll over the new index:
+
+```
+$ invenio index migration rollover "records"
+```
+
+In an in-cluster migration, the rollover works by shuffling around index aliases (since Elasticsearch does not support renaming indexes). In a cross-cluster migration you update the Invenio configuration to point to your new Elasticsearch cluster.
+
+In case something goes wrong during the index migration, you can of course also cancel the job:
+```
+$ invenio index migration cancel "records"
+```
+
+## Bouns
+
+During the sprint a number of other issues were addressed that will also be part of Invenio v3.2:
+
+- **Elasticsearch v7 support** was added to Invenio-Search, Invenio-Indexer and Invenio-Records-REST. Only Invenio-OAIServer needs to be upgraded to support Elasticsearch 7.
+- **Search index prefixing** allowing multiple Invenio instances to use the same Elasticsearch cluster. Elasticsearch by default does not support the concept of virtual hosts. With search indexing prefixing, it will be possible to use a single Elasticsearch cluster for multiple Invenio instances, given that the Invenio instances either trust each other, or you have a protection-layer like Elastic X-Pack or ReadonlyREST in front of Elasticsearch.
```

Comment:
```suggestion
- **Search index prefixing** allowing multiple Invenio instances to use the same Elasticsearch cluster. Elasticsearch by default does not support the concept of virtual hosts. With search index prefixing, it will be possible to use a single Elasticsearch cluster for multiple Invenio instances, given that the Invenio instances either trust each other, or you have a protection-layer like Elastic X-Pack or ReadonlyREST in front of Elasticsearch.
```